dronegen/tag.go

```diff
@@ -0,0 +1,425 @@
+		makefileTarget += "-centos6"
+	}
+	if params.fips {
+		makefileTarget += "-fips"
+	}
+	return makefileTarget
+}
+
+// tagCheckoutCommands builds a list of commands for Drone to check out a git commit on a tag build
+func tagCheckoutCommands(fips bool) []string {
+	commands := []string{
+		`mkdir -p /go/src/github.com/gravitational/teleport`,
+		`cd /go/src/github.com/gravitational/teleport`,
+		`git clone https://github.com/gravitational/${DRONE_REPO_NAME}.git .`,
+		`git checkout ${DRONE_TAG:-$DRONE_COMMIT}`,
+		// fetch enterprise submodules
+		`mkdir -m 0700 /root/.ssh && echo -n "$GITHUB_PRIVATE_KEY" > /root/.ssh/id_rsa && chmod 600 /root/.ssh/id_rsa`,
+		`ssh-keyscan -H github.com > /root/.ssh/known_hosts 2>/dev/null && chmod 600 /root/.ssh/known_hosts`,
+		`git submodule update --init e`,
+		// this is allowed to fail because pre-4.3 Teleport versions don't use the webassets submodule
+		`git submodule update --init --recursive webassets || true`,
+		`rm -f /root/.ssh/id_rsa`,
+		// create necessary directories
+		`mkdir -p /go/cache /go/artifacts`,
+		// set version
+		`if [[ "${DRONE_TAG}" != "" ]]; then echo "${DRONE_TAG##v}" > /go/.version.txt; else egrep ^VERSION Makefile | cut -d= -f2 > /go/.version.txt; fi; cat /go/.version.txt`,
+	}
+	return commands
+}
+
+// tagBuildCommands generates a list of commands for Drone to build an artifact as part of a tag build
+func tagBuildCommands(params tagBuildType) []string {
+	commands := []string{
+		`apk add --no-cache make`,
+		`chown -R $UID:$GID /go`,
+		`cd /go/src/github.com/gravitational/teleport`,
+	}
+
+	if params.fips {
+		commands = append(commands,
+			"export VERSION=$(cat /go/.version.txt)",
+		)
+	}
+
+	commands = append(commands,
+		fmt.Sprintf(
+			`make -C build.assets %s`, tagMakefileTarget(params),
+		),
+	)
+
+	return commands
+}
+
+// tagCopyArtifactCommands generates a set of commands to find and copy built tarball artifacts as part of a tag build
+func tagCopyArtifactCommands(params tagBuildType) []string {
+	extension := ".tar.gz"
+	if params.os == "windows" {
+		extension = ".zip"
+	}
+
+	commands := []string{
+		`cd /go/src/github.com/gravitational/teleport`,
+	}
+
+	// don't copy OSS artifacts for any FIPS build
+	if !params.fips {
+		commands = append(commands,
+			fmt.Sprintf(`find . -maxdepth 1 -iname "teleport*%s" -print -exec cp {} /go/artifacts \;`, extension),
+		)
+	}
+
+	// copy enterprise artifacts
+	if params.os == "windows" {
+		commands = append(commands,
+			`export VERSION=$(cat /go/.version.txt)`,
+			`cp /go/artifacts/teleport-v$${VERSION}-windows-amd64-bin.zip /go/artifacts/teleport-ent-v$${VERSION}-windows-amd64-bin.zip`,
+		)
+	} else {
+		commands = append(commands,
+			`find e/ -maxdepth 1 -iname "teleport*.tar.gz" -print -exec cp {} /go/artifacts \;`,
+		)
+	}
+
+	// we need to specifically rename artifacts which are created for CentOS 6
+	// these is the only special case where renaming is not handled inside the Makefile
```

Comment:
can renaming be handled inside the makefile?

Reply:
Theoretically yes, we could make this change there. As I mentioned above I'm trying to avoid making changes to anything outside of Drone via this PR as much as possible - I'll look to fix these more cosmetic things later as part of a follow-up PR to avoid complications.